pyalgostrategypool/sma_heikin_ashi_crossover.py

```diff
@@ -0,0 +1,209 @@
+        so once a candle starts, strategy_select_instruments_for_exit gets called first
+        and then this method - strategy_select_instruments_for_entry gets called
+        """
+
+        # Add instrument in this bucket if you want to place an order for it
+        # We decide whether to place an instrument in this bucket or not based on the decision making process given below in the loop
+        selected_instruments_bucket = []
+
+        # Add accompanying info in this bucket for that particular instrument in the bucket above
+        sideband_info_bucket = []
+
+        # Looping over all instruments given by you in the bucket (we can give multiple instruments in the configuration)
+        for instrument in instruments_bucket:
+
+            # Compute various things and get the decision to place an order only if no current order is going on (main order is empty / none)
+            if self.main_order.get(instrument) is None:
+
+                if self.main_order.get(instrument) is None:
+
+                    # Get entry decision
+                    action = self.get_decision(instrument)
+                    if action == 'BUY' or (action == 'SELL' and self.strategy_mode is StrategyMode.INTRADAY):
+
+                        # Add instrument to the bucket
+                        selected_instruments_bucket.append(instrument)
+
+                        # Add additional info for the instrument
+                        sideband_info_bucket.append({'action': action})
+
+        # Return the buckets to the core engine
+        # Engine will now call strategy_enter_position with each instrument and its additional info one by one
+        return selected_instruments_bucket, sideband_info_bucket
+
+    def strategy_enter_position(self, candle, instrument, sideband_info):
+        """
+        This method is called once for each instrument from the bucket in this candle
+        place an order here and return it to the core
+        """
+
+        # Quantity formula (number of lots comes from the config)
+        qty = self.number_of_lots * instrument.lot_size
+
+        # TODO: How to set position constants
+        # Place buy order
+        if sideband_info['action'] == 'BUY':
+            self.main_order[instrument] = self.broker.BuyOrderRegular(instrument=instrument, order_code=BrokerOrderCodeConstants.INTRADAY, order_variety=BrokerOrderVarietyConstants.MARKET, quantity=qty)
+            self.profit_order[instrument] = self.broker.SellOrderRegular(instrument=instrument, order_code=BrokerOrderCodeConstants.INTRADAY, order_variety=BrokerOrderVarietyConstants.LIMIT, quantity=qty,
+                                                                         price=self.main_order[instrument].entry_price + self.profit_booking_buy_points,
+                                                                         position=BrokerExistingOrderPositionConstants.EXIT, related_order=self.main_order[instrument])
```

Comment:
`position=BrokerExistingOrderPositionConstants.EXIT`

@pushpakdagade this will work once the pyalgotrading PR is approved